layer/layer_data.h

```diff
@@ -240,6 +240,18 @@ class LayerData {
     return proc_addr;
   }
 
+  // Removes a previously created shader module from the LayerData
+  // This is called while destroying the shader module. Returs the hash
+  // value of the Shader module.
+  uint64_t EraseShader(VkShaderModule shader_module) {
+    uint64_t hash_value = GetShaderHash(shader_module);
+    absl::MutexLock lock(&shader_hash_lock_);
+    auto shader_iter = shader_hash_map_.find(shader_module);
+    assert(shader_iter != shader_hash_map_.end());
```

Comment:
I think this will generate unused variable warnings on non-debug build configurations. You should be able to check for presence/absence with `.count` or `.contains`.